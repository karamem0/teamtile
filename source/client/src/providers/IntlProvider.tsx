//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import {
  IntlProvider as Provider,
  createIntl,
  MessageFormatElement
} from 'react-intl';

import en from '../translations/compiled/en.json';
import ja from '../translations/compiled/ja.json';

const translations: { [key: string]: Record<string, MessageFormatElement[]> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language.substring(0, 2)]
});

function IntlProvider(props: React.PropsWithChildren<unknown>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
