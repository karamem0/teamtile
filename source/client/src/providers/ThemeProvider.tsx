//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { Provider, themeNames } from '@fluentui/react-teams';

import { app } from '@microsoft/teams-js';

import { inTeams } from '../utils/Teams';

interface ThemeProviderProps {
  children?: React.ReactNode
}

function ThemeProvider(props: ThemeProviderProps) {

  const { children } = props;

  const [ lang, setLang ] = React.useState<string>('en-US');
  const [ themeName, setThemeName ] = React.useState<themeNames>(themeNames.Default);

  const handleThemeChange = (value: string) => {
    switch (value) {
      case 'dark':
        setThemeName(themeNames.Dark);
        break;
      case 'contrast':
        setThemeName(themeNames.HighContrast);
        break;
      default:
        setThemeName(themeNames.Default);
        break;
    }
  };

  React.useEffect(() => {
    if (!inTeams()) {
      return;
    }
    (async () => {
      await app.initialize();
      const context = await app.getContext();
      setLang(context.app.locale);
      handleThemeChange(context.app.theme);
      app.registerOnThemeChangeHandler(handleThemeChange);
    })();
  }, []);

  return (
    <Provider
      lang={lang}
      themeName={themeName}>
      {children}
    </Provider>
  );

}

export default ThemeProvider;
