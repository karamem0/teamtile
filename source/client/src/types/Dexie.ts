//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

export interface ArrayEntity<T> {
  id: string,
  expired: number,
  values: T[]
}

export interface ValueEntity<T> {
  id: string,
  expired: number,
  value: T
}
