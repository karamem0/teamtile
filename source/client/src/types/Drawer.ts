//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

export type DrawerType =
  | 'channel'
  | 'member'
  | 'tag';
