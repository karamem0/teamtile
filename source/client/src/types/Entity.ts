//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

export enum MembershipType {
  private = 'private',
  standard = 'standard'
}

export enum VisibilityType {
  private = 'private',
  public = 'public'
}

export enum AccordionType {
  all = 'all',
  archived = 'archived',
  pinned = 'pinned'
}

export interface Channel {
  id?: string,
  displayName?: string,
  membershipType?: MembershipType,
  webUrl?: string
}

export interface Drive {
  id?: string,
  webUrl?: string
}

export interface Group {
  id?: string,
  email?: string,
  sensitivityLabel?: string
}

export interface Icon {
  id?: string,
  data?: string
}

export interface Member {
  id?: string,
  displayName?: string,
  email?: string,
  icon?: string,
  userId?: string
}

export interface Tab {
  id?: string,
  appId?: string,
  displayName?: string,
  webUrl?: string
}

export interface Team {
  id?: string,
  archived?: boolean,
  description?: string,
  displayName?: string,
  icon?: string,
  internalId?: string,
  visibility?: VisibilityType,
  webUrl?: string
}

export interface TeamInfo {
  id?: string,
  displayName?: string,
  tenantId?: string
}
