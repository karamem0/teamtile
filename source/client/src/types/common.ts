//
// Copyright (c) 2021 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/master/LICENSE
//

export interface KeyValue<K, V> {
  key: K,
  value: V
}
