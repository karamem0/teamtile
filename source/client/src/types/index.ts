//
// Copyright (c) 2021 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/master/LICENSE
//

export interface Team {
  id?: string;
  internalId?: string;
  name?: string;
  description?: string;
  visibility?: string;
  url?: string;
  icon?: string;
}

export interface TeamChannels {
  count?: number;
  nextLink?: string;
  values?: TeamChannel[];
}

export interface TeamChannel {
  id?: string;
  name?: string;
  url?: string;
}

export interface TeamMembers {
  count?: number;
  nextLink?: string;
  values?: TeamMember[];
}

export interface TeamMember {
  id?: string;
  name?: string;
  email?: string;
}

export interface TeamDrive {
  id?: string;
  url?: string;
}
