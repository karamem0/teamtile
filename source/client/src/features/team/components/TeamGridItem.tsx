//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { TeamCard } from '../../../types/Store';
import { app } from '@microsoft/teams-js';

import Presenter from './TeamGridItem.presenter';

export interface TeamGridItemProps {
  card: TeamCard
}

function TeamGridItem(props: Readonly<TeamGridItemProps>) {

  const { card } = props;

  const handleClick = React.useCallback(() => {
    if (card.team.webUrl == null) {
      return;
    }
    app.openLink(card.team.webUrl);
  }, [
    card
  ]);

  return (
    <Presenter
      card={card}
      onClick={handleClick} />
  );

}

export default TeamGridItem;
