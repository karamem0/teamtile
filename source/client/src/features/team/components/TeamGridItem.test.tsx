//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import IntlProvider from '../../../providers/IntlProvider';
import TeamGridItem from './TeamGridItem.presenter';
import ThemeProvider from '../../../providers/ThemeProvider';
import { VisibilityType } from '../../../types/Entity';
import { render } from '@testing-library/react';

jest.mock('./AvatarIcon', () =>
  function AvatarIcon({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="AvatarIcon">
        {children}
      </div>
    );
  });

jest.mock('./CalendarMenuItem', () =>
  function CalendarMenuItem({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="CalendarMenuItem">
        {children}
      </div>
    );
  });

jest.mock('./ChannelMenuItem', () =>
  function ChannelMenuItem({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="ChannelMenuItem">
        {children}
      </div>
    );
  });

jest.mock('./DriveMenuItem', () =>
  function DriveMenuItem({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="DriveMenuItem">
        {children}
      </div>
    );
  });

jest.mock('./MemberMenuItem', () =>
  function MemberMenuItem({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="MemberMenuItem">
        {children}
      </div>
    );
  });

jest.mock('./PinMenuItem', () =>
  function PinMenuItem({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="PinMenuItem">
        {children}
      </div>
    );
  });

jest.mock('./SensitivityLabel', () =>
  function SensitivityLabel({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="SensitivityLabel">
        {children}
      </div>
    );
  });

jest.mock('./TagMenuItem', () =>
  function TagMenuItem({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="TagMenuItem">
        {children}
      </div>
    );
  });

jest.mock('./VisibilityIcon', () =>
  function VisibilityIcon({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="VisibilityIcon">
        {children}
      </div>
    );
  });

it('should create a shapshot when loading is true', () => {
  const params = {
    card: {
      id: '02bd9fd6-8f93-4758-87c3-1fb73740a315',
      team: {
        id: '02bd9fd6-8f93-4758-87c3-1fb73740a315',
        archived: false,
        displayName: 'HR Taskforce',
        description: 'Welcome to the HR Taskforce team.',
        internalId: '19:09fc54a3141a45d0bc769cf506d2e079@thread.skype',
        sensitivityLabel: 'Restricted',
        visibility: 'private' as VisibilityType,
        webUrl: 'https://teams.microsoft.com/l/team/19:09fc54a3141a45d0bc769cf506d2e079%40thread.skype/conversations?groupId=02bd9fd6-8f93-4758-87c3-1fb73740a315&tenantId=dcd219dd-bc68-4b9b-bf0b-4a33a796be35'
      },
      loading: true,
      visible: true
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <TeamGridItem {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should create a shapshot when loading is false', () => {
  const params = {
    card: {
      id: '02bd9fd6-8f93-4758-87c3-1fb73740a315',
      team: {
        id: '02bd9fd6-8f93-4758-87c3-1fb73740a315',
        archived: false,
        displayName: 'HR Taskforce',
        description: 'Welcome to the HR Taskforce team.',
        internalId: '19:09fc54a3141a45d0bc769cf506d2e079@thread.skype',
        sensitivityLabel: 'Restricted',
        visibility: 'private' as VisibilityType,
        webUrl: 'https://teams.microsoft.com/l/team/19:09fc54a3141a45d0bc769cf506d2e079%40thread.skype/conversations?groupId=02bd9fd6-8f93-4758-87c3-1fb73740a315&tenantId=dcd219dd-bc68-4b9b-bf0b-4a33a796be35'
      },
      loading: false,
      visible: true
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <TeamGridItem {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
