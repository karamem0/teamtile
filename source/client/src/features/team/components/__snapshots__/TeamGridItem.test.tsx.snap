// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot when the loading is false 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  line-height: 1.25em;
  background-color: #f5f5f5;
}

.emotion-1 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto 1fr;
  grid-gap: 0.5rem;
}

.emotion-2 {
  display: grid;
  grid-template-rows: auto auto auto;
  grid-template-columns: 1fr auto;
  grid-gap: 0.25rem;
}

.emotion-3 {
  grid-row: 1/2;
  grid-column: 1/2;
  height: 1.5rem;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-4 {
  font-weight: 600;
  color: #4c6978;
  white-space: nowrap;
  cursor: pointer;
}

.emotion-5 {
  grid-row: 1/3;
  grid-column: 2/3;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-row: 2/3;
  grid-column: 1/2;
  grid-gap: 0.25rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -ms-flex-pack: left;
  -webkit-justify-content: left;
  justify-content: left;
  height: 1.25rem;
}

.emotion-7 {
  grid-row: 3/4;
  grid-column: 1/3;
  height: 6.25rem;
}

.emotion-8 {
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-line-clamp: 5;
  line-height: 1.25rem;
  -webkit-box-orient: vertical;
}

<div
    class="fui-FluentProvider fui-FluentProviderr1 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    >
      <div
        class="fui-Card rfxo2k2 ___1x1hcrp_1926re5 f1vx9l62 f14k419y f1fgo9fz fvqmfsm f3am6yf f1r5wgso frsmuga fuldkky fxugw4r f1whvlc6 f16gxe2i fpgykix f1osi826 fzybk4o ftqa4ok f2hkw1w f8hki3x f1d2448m f1bjia2o ffh67wi f226i61 f13kzufm flujwa2 fsx75g8 f15bsgw9 f14e48fq f18yb2kv fd6o370 fpqizxz fnd8nzh f15fr7a0 fwsq40z f34ld9f fy0y4wt"
        role="listitem"
      >
        <div
          class="emotion-1"
        >
          <div
            data-testid="test-AvatarIcon"
          />
          <div
            class="emotion-2"
          >
            <div
              class="emotion-3"
            >
              <span
                class="fui-Text emotion-4 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
                role="button"
              >
                HR Taskforce
              </span>
            </div>
            <div
              class="emotion-5"
            >
              <button
                aria-haspopup="menu"
                class="fui-Button r1alrhcs ___cpe5h10_1yhfyj7 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3 f18ktai2 fwbmr0d f44c6la"
                id="menur2"
                type="button"
              >
                <span
                  class="fui-Button__icon rywnvv2"
                >
                  <svg
                    aria-hidden="true"
                    class="___12fm75w_1fmvgpr f1w7gpdv fez10in fg4l7m0"
                    fill="currentColor"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 5a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm-1 3a1 1 0 1 0 2 0 1 1 0 0 0-2 0Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="emotion-6"
            >
              <div
                data-testid="test-VisibilityIcon"
              />
              <div
                data-testid="test-SensitivityLabel"
              />
              <div />
            </div>
            <div
              class="emotion-7"
            >
              <div
                class="emotion-8"
              >
                Welcome to the HR Taskforce team.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot when the loading is true 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  line-height: 1.25em;
  background-color: #f5f5f5;
}

.emotion-1 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto 1fr auto;
  grid-gap: 0.5rem;
}

.emotion-2 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto;
  grid-gap: 0.25rem;
}

<div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    >
      <div
        class="fui-Card rfxo2k2 ___1x1hcrp_1926re5 f1vx9l62 f14k419y f1fgo9fz fvqmfsm f3am6yf f1r5wgso frsmuga fuldkky fxugw4r f1whvlc6 f16gxe2i fpgykix f1osi826 fzybk4o ftqa4ok f2hkw1w f8hki3x f1d2448m f1bjia2o ffh67wi f226i61 f13kzufm flujwa2 fsx75g8 f15bsgw9 f14e48fq f18yb2kv fd6o370 fpqizxz fnd8nzh f15fr7a0 fwsq40z f34ld9f fy0y4wt"
        role="listitem"
      >
        <div
          aria-busy="true"
          class="fui-Skeleton"
          role="progressbar"
        >
          <div
            class="emotion-1"
          >
            <div
              class="fui-SkeletonItem ___6z4oc50_11iqglg f10pi13n f1a3p1vp f13zj6fq f2gz7yw f1mdlcz9 f1nt53hs f18xs7ez fs3pcyf f1f9zsvm f1ub80pn f1e7twep f171zwvm f1yu5riq f5q2cvs f124akge ff2sm71"
            />
            <div
              class="emotion-2"
            >
              <div
                class="fui-SkeletonItem ___148mfmj_1esmiem f10pi13n f1a3p1vp f13zj6fq f2gz7yw f1mdlcz9 f1nt53hs f18xs7ez fs3pcyf f1f9zsvm f1ub80pn f1e7twep f171zwvm f1yu5riq fly5x3f ff3glw6 frvgh55"
              />
              <div
                class="fui-SkeletonItem ___148mfmj_1esmiem f10pi13n f1a3p1vp f13zj6fq f2gz7yw f1mdlcz9 f1nt53hs f18xs7ez fs3pcyf f1f9zsvm f1ub80pn f1e7twep f171zwvm f1yu5riq fly5x3f ff3glw6 frvgh55"
              />
              <div
                class="fui-SkeletonItem ___1316iuv_vy383s0 f10pi13n f1a3p1vp f13zj6fq f2gz7yw f1mdlcz9 f1nt53hs f18xs7ez fs3pcyf f1f9zsvm f1ub80pn f1e7twep f171zwvm f1yu5riq fly5x3f ff3glw6 fypu0ge"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
